backend/ventserver/protocols/file.py

```diff
@@ -0,0 +1,147 @@
+"""Sans-I/O protobuf file handling protocol."""
+
+import logging
+from typing import Optional
+
+import attr
+import betterproto
+
+from ventserver.protocols import mcu
+from ventserver.protocols import exceptions
+from ventserver.protocols import messages
+from ventserver.protocols import crcelements
+from ventserver.sansio import protocols
+from ventserver.sansio import channels
+
+
+# Classes
+
+@attr.s(auto_attribs=True)
+class StateData:
```

Comment:
I believe this class should inherit from `protocols.events.Event`? Then we can use the `has_data` method to check inputs in the `FileReceiver.input` method.

Reply:
You  marked this resolved but I don't see any difference in the code here - is there a commit you forgot to push up, or are you thinking `StateData` shouldn't inherit from `protocols.events.Event`?